app/screens/login.tsx

```diff
@@ -1,45 +1,46 @@
 import React, { useEffect } from "react";
 import { observer } from 'mobx-react-lite';
-import { StyleSheet, View } from 'react-native';
+import { StyleSheet, View, Text, Alert } from 'react-native';
 import { NavigationActions, NavigationInjectedProps } from 'react-navigation';
 import { Toggle } from 'react-powerplug';
 import { color, spacing } from '../theme';
-import { useStores } from '../models/root-store';
 import PromiseTimeout from '../services/timers/promise-timeout.service';
 import { Button, Checkbox, Screen, Icon, TextField } from '../components';
 import ErrorModal from '../components/modal-error/error.modal';
 import LoadingModal from '../components/loading/loading.modal';
+import { useStores } from "../models/root-store";
 
-export interface LoginProps extends NavigationInjectedProps<{}> {};
+export interface LoginProps extends NavigationInjectedProps<{}> {}
 
+const ERROR_MESSAGE = 'אירעה שגיאה.\nאנה נסה שוב מאוחר יותר.';
 export const LoginScreen: React.FunctionComponent<LoginProps> = observer(() => {
-  const { navigationStore, profileModel, profileModel: { login, token, autoLogin, silentLogin } } = useStores();
+  const { navigationStore, profileModel: { login, token, autoLogin, silentLogin } } = useStores();
 
-  const [username, setUserName] = React.useState<string>(__DEV__ ? 'haver4' : '');
-  const [password, setPassword] = React.useState<string>(__DEV__ ? '0545606871' : '');
-  const [isErrorModalDisplayed, setErrorModalState] = React.useState<boolean>(false);
-  const [isLoadingModalDisplayed, setLoadingModalState] = React.useState<boolean>(false);
+  const [username, setUserName] = React.useState<string>(__DEV__ ? 'haver3' : '');
+  const [password, setPassword] = React.useState<string>(__DEV__ ? '0545718472' : '');
+  const [isErrorModalDisplayed, setErrorModal] = React.useState<boolean>(false);
+  const [loginError, setLoginError] = React.useState<boolean>(false);
+  const [isLoadingModalDisplayed, setLoadingModal] = React.useState<boolean>(false);
   const [savePass, setSavePass] = React.useState<boolean>(autoLogin);
   const [errorMessage, setErrorMessage] = React.useState<string>(undefined);
 
   useEffect(() => {
     if (autoLogin && token) {
       displayLoadingModal(true);
-      silentLogin().then(r => {
-        handleLoginRequest(r);
+      silentLogin().then(response => {
+        handleLoginRequest(response);
       });
     }
   }, []);
 
-  const displayErrorModal = React.useCallback(() => {
-    setErrorModalState(!isErrorModalDisplayed);
-  }, [isErrorModalDisplayed]);
-
-  const displayLoadingModal = (show: boolean) => {
-    setLoadingModalState(show);
+  const displayErrorModal = () => {
+	setErrorMessage(ERROR_MESSAGE);
+	setErrorModal(!isErrorModalDisplayed);
   };
 
- 	const loginSequence = async () => {
+  const displayLoadingModal = (show: boolean) => setLoadingModal(show);
+
+  const loginSequence = async () => {
```

Comment:
PromiseTimout below with try-catch is redundant because we use apiSauce (Axios) which can handle this for us.